.github/workflows/reusable-acceptance.yml

```diff
@@ -0,0 +1,128 @@
+        required: false 
+        type: string
+        default: docker.mirror.hashicorp.services/hashicorpdev/consul-k8s-control-plane:latest
+      directory:
+        required: true
+        type: string
+      go-version:
+        required: true
+        type: string
+      gotestsum-version:
+        required: true
+        type: string
+      kind-version:
+        required: false
+        type: string
+        default: "v1.22.4"
+    secrets:
+        CONSUL_ENT_LICENSE:
+          required: true
+
+# Environment variables can only be used at the step level
+env:
+  TEST_RESULTS: /tmp/test-results # path to where test results are saved
+  CONSUL_ENT_LICENSE: ${{ secrets.CONSUL_ENT_LICENSE }} 
+
+jobs:
+  a:
+    runs-on: ubuntu-latest
+    strategy: 
+      matrix:
+        include: # I am really sorry for this but I could not find a way to automatically split our tests into several runners. For now, split manually.
+          - {runner: "0", test-packages: "basic connect consul-dns"}
+          - {runner: "1", test-packages: "controller example ingress-gateway"}
+          - {runner: "2", test-packages: "mesh-gateway metrics"}
+          - {runner: "3", test-packages: "partitions sync terminating-gateway"}
+          - {runner: "4", test-packages: "vault"}
+
+      fail-fast: true
+    steps:      
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.go-version }}
+
+      - name: Setup go mod cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install gotestsum
+        run: |
+          wget https://github.com/gotestyourself/gotestsum/releases/download/v"${{ inputs.gotestsum-version }}"/gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+          sudo tar -C /usr/local/bin -xzf gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+          rm gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+
+      - run: mkdir -p ${{ env.TEST_RESULTS }}
+
+      - name: go mod download
+        working-directory: ${{ inputs.directory }}
+        run: go mod download
+      
+      - name: Create kind clusters
+        run: |
+          kind create cluster --name dc1 --image kindest/node:${{ inputs.kind-version }}
+          kind create cluster --name dc2 --image kindest/node:${{ inputs.kind-version }}
+
+      # We have to run the tests for each package separately so that we can
+      # exit early if any test fails (-failfast only works within a single
+      # package).
+      - name: Run acceptance tests ${{ matrix.runner }}
+        working-directory: ${{ inputs.directory }}
+        if: github.repository_owner == 'hashicorp' # This prevents running on forks
```

Comment:
`if: github.repository_owner == 'hashicorp'` prevents the tests from running on forks.